### PR TITLE
Refactor CoverageStatistics class & specs and fix typos

### PR DESCRIPTION
**Pull request by ddzz**
Simple refactor of `SimpleCov::CoverageStatistics` that makes the code slightly more efficient and readable. Also fixes a spec (along with some clean up).

I also searched for typos using [cspell](https://www.npmjs.com/package/cspell). I manually fixed the obvious typos that were reported (and ignored tons of false positive matches).

### CODE REVIEW

File: lib/simplecov/coverage_statistics.rb [hunk 1]
```diff
@@ -35,22 +35,22 @@ def initialize(covered:, missed:, total_strength: 0.0)
...
+      return 100.0 if @missed.zero?
 
-      covered * 100.0 / total
+      covered * 100.0 / @total
     end
 
-    def compute_strength(total_strength, total)
-      return 0.0 if total.zero?
+    def compute_strength(total_strength)
+      return 0.0 if @total.zero?
```
Comment (PragTob) [inline lib/simplecov/coverage_statistics.rb]: The usage of parameters here (and rest of the file) was intentional. That way it's clearer to me what part of the computation relies on which values just by looking at the method signature/seeing their calls in initialize :)

I know one could debate it isn't very OO and if they were public it wouldn't be like this, but personally I like them better like this :)

File: spec/coverage_statistics_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 require "helper"
 
 RSpec.describe SimpleCov::CoverageStatistics do
-  describe ".new" do
+  describe "::new" do
```
Comment (PragTob) [inline spec/coverage_statistics_spec.rb]: ? That's not the convention I usually follow in Ruby. For me it's `.class_method` and `#instance_method` :)

File: spec/coverage_statistics_spec.rb [hunk 3]
```diff
@@ -16,20 +16,20 @@
     end
 
     it "can omit the total strength defaulting to 0.0" do
-      statistics = described_class.new(covered: 4, missed: 6, total_strength: 0.0)
+      statistics = described_class.new(covered: 4, missed: 6)
```
Comment (PragTob) [inline spec/coverage_statistics_spec.rb]: :+1:

File: spec/coverage_statistics_spec.rb [hunk 4]
```diff
@@ -16,20 +16,20 @@
...
 
       expect(statistics.strength).to eq 0.0
     end
 
     it "can deal with it if everything is 0" do
-      statistics = described_class.new(covered: 0, missed: 0, total_strength: 0.0)
+      statistics = empty_statistics
```
Comment (PragTob) [inline spec/coverage_statistics_spec.rb]: technically yes, while the test description says "everything is 0" I like to see that everything is 0 that makes it immedeatly a bit clearer (for this one explicit test) and it's not such a huge amount of duplication imo.

File: spec/coverage_statistics_spec.rb (hunk 2 above)
Comment (ddzz) [reply spec/coverage_statistics_spec.rb]: The `::` convention is used [in Ruby Docs](https://ruby-doc.org/core-2.2.0/BasicObject.html) for public class methods (see the list of methods on the left), but I do think the RSpec convention is to use `.`. I'll change it back. Thanks for reviewing!

**Pull request merged**